apps/gateway/src/worker.ts

```diff
@@ -21,6 +21,30 @@ export async function processLogQueue(): Promise<void> {
 				...i,
 			})),
 		);
+
+		for (const data of logData) {
+			if (!data.cost || data.cached) {
+				continue;
+			}
+
+			const project = await db.query.project.findFirst({
+				where: {
+					id: {
+						eq: data.projectId,
+					},
+				},
+			});
```

Comment:
create a shared helper function for this, which additionally caches the project for 60 seconds for better perf